### DISCUSSION TITLE
Maintenance Mode active, app still trying to write to DB

### DISCUSSION THREAD

**Discussion by oversize** [netbox-community/netbox#6219] [Q&A] [answered]
We recently wanted to migrate installations and therfore wanted to enable "readonly" mode. For this i have done two things:
First, i enabled MAINTENANCE_MODE in the config.py and restartet the service. I can see the maintenance banner and thus assume its configured correctly. Then, i set the database to readonly using `alter database netbox set default_transaction_read_only=on;` .

Unfortunatly, when i log in it seems as if the app still wants to write to the database. I the following error:

> <class 'django.db.utils.InternalError'>
>FEHLER:  UPDATE kann nicht in einer Read-Only-Transaktion ausgeführt werden

which basically translates to "ERROR: Cant Update in Readonly mode"  (because i have told the db to be readonly).

I am using:
* Postgresql: 11
* Python version: 3.7.3
* NetBox version: 2.11.0

So, I am wondering, What am i missing (doing wrong) to enable a readonly mode for netbox? 

Thank you
Manuel

**Comment by candlerb** [comment, marked as answer]
This came up before: see [google groups](https://groups.google.com/g/netbox-discuss/search?q=cannot%20execute%20UPDATE%20in%20a%20read-only%20transaction).

In short, when you login, by default Netbox/Django updates the sessions table, so it doesn't work against a read-only database or replica.

Using [SESSION_FILE_PATH](https://netbox.readthedocs.io/en/stable/configuration/optional-settings/#session_file_path) (see #2426) might work.  There was also a feature request for cookie-based sessions (#3050, #3118) and there's more info in those tickets.

**Reply by oversize** [reply]
Ah ok, thank you for the heads up and the links,. Will have a look into them and see if moving the session store out of the database is an option for us. The Documentation of MAINTENANCE_MODE suggests that read-only mode supports logins though: 

> This is to allow new logins when the database is in a read-only state. 

https://netbox.readthedocs.io/en/stable/configuration/optional-settings/#maintenance_mode
